hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -1721,7 +1722,7 @@ public String generateSelectVersionString() {
 		SimpleSelect select = new SimpleSelect( getFactory().getDialect() )
 				.setTableName( getVersionedTableName() );
 		if ( isVersioned() ) {
-			select.addColumn( versionColumnName );
+			select.addColumn( getVersionColumnName(), VERSION_COLUMN_ALIAS );
 		}
 		else {
 			select.addColumns( rootTableKeyColumnNames );
```

Comment:
Assuming this block ever happens, wouldn't you need to apply the same alias here so the read from ResultSet later can find it?

Reply:
To be clear, 

> I did look into that, but apparently that use case is broken in a different way (NPE is thrown by EntityVerifyVersionProcess#doBeforeTransactionCompletion).

Reply:
Yes, my guess is it fails trying to compare the versions right.  I.e. `entry.getVersion()` returns null, so that `entry.getVersion().equals()`.

My point is that none of this should be happening if the entity is not versioned.  If that is what you mean by "broken in a different way", then sure.

Reply:
Yes, you're correct about the NPE, and that it won't happen if the entity is versioned.

What I meant by "broken in a different way", was that the non-versioned use case is broken by the NPE before being broken by the lack of an alias for a quoted column.